File: backend/src/main/java/org/example/backend/recipe/model/RecipeModel.java
package org.example.backend.recipe.model;

import java.util.*;

public record RecipeModel(
        String title,
        String id,
        List<IngredientModel> ingredients,
        NutritionValuesModel nutritionValues,
        int totalTime,
        Optional<List<String>> allergens,
        List<String> tags,
        String image
) {
}

File: frontend/src/models/recipe/RecipeModel.ts
import type { IngredientGroupModel } from "./IngredientGroupModel.ts";
import type { NutritionValuesModel } from "./NutritionValuesModel.ts";

export type RecipeModel = {
  id: string;
  title: string;
  ingredients: IngredientGroupModel[];
  nutritionValues: NutritionValuesModel;
  totalTime: number;
  allergens: string[] | null;
  tags: string[];
  image: string;
};
